### PR TITLE
MailGuard 365 Sentinel Solution

### PR DESCRIPTION
**Pull request by prathikc**
Required items, please complete
   
   Change(s):
   - MailGuard 365 Solution for Sentinel - data connector, hunting queries, workbook and logo.

   Reason for Change(s):
   - New sentinel connector for MailGuard 365. 

   Version Updated:
   - N/A

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by prathikc** [comment]
@microsoft-github-policy-service agree company="MailGuard Pty Ltd"

**Comment by v-rbajaj** [comment]
Hi @devikamehra, please provide your feedback.

**Review by devikamehra — CHANGES_REQUESTED** [review]
Please provide sample data to test Hunting queries.

**Comment by v-rbajaj** [comment]
Hi @prathikc, please address the comments from me and Devika, thanks!

File: Solutions/MailGuard 365/Hunting Queries/MailGuard365HighConfidenceThreats.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  - connectorId: MailGuard365
+    dataTypes:
+      - MailGuard365
+tactics:
+  - Reconnaissance
+relevantTechniques:
+  - T1598
+query: |
+  MailGuard365_Threats_CL
+  | where Score_d > 20
```
[the remark this answers is not in the document]
Comment (prathikc) [reply Solutions/MailGuard 365/Hunting Queries/MailGuard365HighConfidenceThreats.yaml:16]: I have updated the queries to extend the entity mapping.

File: .script/tests/KqlvalidationsTests/CustomTables/MailGuard365_Threats_CL.json [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        {
+            "Name": "Score_d",
+            "Type": "Real"
+        },
+        {
+            "Name": "Virus_b",
+            "Type": "Bool"
+        },
+        {
+            "Name": "Category",
```
[the remark this answers is not in the document]
Comment (prathikc) [reply .script/tests/KqlvalidationsTests/CustomTables/MailGuard365_Threats_CL.json:41]: Yes, i have checked the schema again and that field is without _s.

**Comment by v-rbajaj** [comment]
> Hi @prathikc, please provide the Sample data. Also, repackage the solution using this tool https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V2 with version 2.0.0

Hi @prathikc, can you please work on this as well?

**Comment by prathikc** [comment]
> > Hi @prathikc, please provide the Sample data. Also, repackage the solution using this tool https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V2 with version 2.0.0
> 
> Hi @prathikc, can you please work on this as well?

Hi @v-rbajaj , I have uploaded the sample data as well now. Running into a few issues with generating the solution package, i suspect it might be with my environment, will continue to work on it.

**Comment by prathikc** [comment]
Hi @v-rbajaj , I've updated the solution V2.

File: Solutions/MailGuard 365/Data Connectors/MailGuard365.json [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        {
+            "metricName": "Total data received",
+            "legend": "MailGuard365_Threats_CL",
+            "baseQuery": "MailGuard365_Threats_CL"
+        }
+    ],
+    "sampleQueries": [
+        {
+            "description" : "All phishing threats stopped by MailGuard 365",
+            "query": "MailGuard365_Threats_CL \n | where where Category == \"Phishing\""
```
Comment (v-rbajaj) [inline Solutions/MailGuard 365/Data Connectors/MailGuard365.json:16]: Please remove the extra "where" form the query, this should fix the failing validation.

Comment (v-rbajaj) [reply Solutions/MailGuard 365/Data Connectors/MailGuard365.json:16]: Please do fix the ARM TTK validation as well.

Comment (prathikc) [reply Solutions/MailGuard 365/Data Connectors/MailGuard365.json]: Thank you, I've fixed the KQL validation and the ARM TTK validation as well.

**Comment by v-rbajaj** [comment]
Hi @prathikc, I am still not able to test the maintemplate. I am gettting this error message. My wild guess is there should be workbook version and id in variables which are currently blank.
image

**Comment by prathikc** [comment]
> Hi @prathikc, I am still not able to test the maintemplate. I am gettting this error message. My wild guess is there should be workbook version and id in variables which are currently blank. image

Hi @v-rbajaj - apologies for the delay here as i am currently on leave. I've updated the version id and content id, hope it helps get past the validation.

**Review by devikamehra — APPROVED** [review]
Validated only the hunting queries. They look good.

**Comment by v-rbajaj** [comment]
Hi @prathikc, Thanks for making the changes but I am still not able to deploy the maintemplate.
Can you check from your side if there is something missing?
image

**Comment by prathikc** [comment]
Hi @v-rbajaj, I've reviewing the template again, do you have any ideas as to which resource is missing a description parameter?

**Comment by v-rbajaj** [comment]
Hi @prathikc, since you have added the a new workbook, please update the Workbook metadata here --> https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json

**Comment by v-rbajaj** [comment]
Hi @prathikc, can you please make the changes as requested above?

**Comment by v-rbajaj** [comment]
Hi @prathikc, can you please act on above comments?

**Comment by devikamehra** [comment]
@vakohl PR is approved for Hunting Queries. Only tagging for visibility

**Comment by v-rbajaj** [comment]
Hi @prathikc, can you please act on above comments?

**Comment by prathikc** [comment]
> Hi @prathikc, can you please act on above comments?

Hi @v-rbajaj, Apologies for the delay in response, i have been away. Will look into updating and packaging the solution.

**Comment by v-rbajaj** [comment]
Sure, thanks @prathikc.

**Comment by v-rbajaj** [comment]
Hi @prathikc, thanks for making these changes.
As per latest guidelines you would repackage the solution using this tool https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V3
Please go through the readme.md file and follow the instructions mentioned there.

Please remove the 2.0.0 zip from this PR, this PR will only have 3.0.0 zip

Please do add release note as well, follow below mentioned link for the same.
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/ReleaseNotesGuidance.md

**Comment by v-rbajaj** [comment]
> Hi @prathikc, thanks for making these changes. As per latest guidelines you would repackage the solution using this tool https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V3 Please go through the readme.md file and follow the instructions mentioned there.
> 
> Please remove the 2.0.0 zip from this PR, this PR will only have 3.0.0 zip
> 
> Please do add release note as well, follow below mentioned link for the same. https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/ReleaseNotesGuidance.md

Hi @prathikc, please look into this comment. 
Also please fix the merge conflicts.

**Comment by v-rbajaj** [comment]
Hi @prathikc , please look into the above comment.

**Comment by prathikc** [comment]
> Hi @prathikc , please look into the above comment.

@v-rbajaj Noted, will package in V3.

**Comment by v-atulyadav** [comment]
Noted @prathikc, thanks.

**Comment by v-rbajaj** [comment]
Hi @prathikc, can you please provide some update on this PR?

**Comment by v-rbajaj** [comment]
Hi @prathikc, can you please provide some update on this PR?

**Comment by v-rbajaj** [comment]
Hi @prathikc, please provide some update on this PR

**Comment by prathikc** [comment]
> Hi @prathikc, please provide some update on this PR

Hi @v-rbajaj , i am working on packaging the solution. There are some urgent issues that I am currently working on, will provi

> Hi @prathikc, please provide some update on this PR

Hi @v-rbajaj , I'm still working on the packaging, will get it ready by early next week.

**Comment by v-rbajaj** [comment]
Hi @prathikc, thanks noted.

**Comment by v-rbajaj** [comment]
Hi @prathikc, can you please provide some update on this PR?

**Comment by v-rbajaj** [comment]
Hi @prathikc, thanks for making these changes, I hope you are working on adding of release notes and fixing the merge conflicts?

**Comment by prathikc** [comment]
> Hi @prathikc, thanks for making these changes, I hope you are working on adding of release notes and fixing the merge conflicts?

Hi @v-rbajaj , i have resolved the conflicts and have added in release notes.

File: Solutions/MailGuard 365/Data Connectors/MailGuard365.json [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                "requiredPermissions": {
+                    "action": true
+                }
+            }
+        ]
+    },
+    "instructionSteps": [
+        {
+            "title": "Configure and connect MailGuard 365",
+            "description": "1. In the MailGuard 365 Console, click **Settings** on the navigation bar.\n2. Click the **Integrations** tab.\n3. Click the **Enable Microsoft Sentinel**.\n  - Enter your workspace id and primary key from the fields below, click **Finish**.\n5. For additional instructions, please contact MailGuard 365 support.",
```
Comment (v-rbajaj) [inline Solutions/MailGuard 365/Data Connectors/MailGuard365.json:72]: Hi @prathikc, the description needs to be fixed,  there is a small problem with the numbering, please take a look at image below.
image

Please repackage the solution post fixing.

Comment (prathikc) [reply Solutions/MailGuard 365/Data Connectors/MailGuard365.json]: Hi @v-rbajaj , that was a good catch! I have made the changes and re-packaged the solution.

**Comment by v-rbajaj** [comment]
Hi @prathikc,  I just wanted to confirm if "Enter your workspace id and primary key from the fields below, click Finish." should be point 4 and "For additional instructions, please contact MailGuard 365 support." should be 5? Asking because it does look a bit odd..
image

File: Solutions/MailGuard 365/SolutionMetadata.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+    "publisherId": "mailguardptylimited",
+    "offerId": "mailguard365",
```
Comment (v-rbajaj) [inline Solutions/MailGuard 365/SolutionMetadata.json:3]: Apart from that you would need to change the `offerId `

You can refer this document  for the same - https://github.com/Azure/Azure-Sentinel/tree/master/Solutions#author-and-publish-a-solution-offer

Please repackage it after making this change.

Comment (prathikc) [reply Solutions/MailGuard 365/SolutionMetadata.json]: Hi @v-rbajaj  - I have made both the requested changes.

**Review by v-rbajaj — APPROVED** [review]

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**